src/cpp/src/module_genai/modules/md_text_embedding.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2023-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "md_text_embedding.hpp"
+#include "visual_language/embedding_model.hpp"
+#include "circular_buffer_queue.hpp"
+
+namespace ov {
+namespace genai {
+namespace module {
+
+void TextEmbeddingModule::print_static_config() {
+    std::cout << R"(
+  text_embedding:                     # Module Name
+    type: "TextEmbeddingModule"
+    description: "Compute text embeddings from token ids."
+    device: "GPU"
+    inputs:
+      - name: "input_ids"
+        type: "OVTensor"              # Token ids tensor from TextEncoderModule
+        source: "ParentModuleName.input_ids"
+    outputs:
+      - name: "input_embedding"
+        type: "OVTensor"              # Or OVRemoteTensor depending on device / use case
+    params:
+      model_path: "path/to/text_embeddings_model_dir"
+      scale_emb: "1.0"
+    )" << std::endl;
+}
+
+TextEmbeddingModule::TextEmbeddingModule(const IBaseModuleDesc::PTR& desc) : IBaseModule(desc) {
+    if (!initialize()) {
+        std::cerr << "Failed to initiate TextEmbeddingModule" << std::endl;
+    }
+}
+
+bool TextEmbeddingModule::initialize() {
+    const auto& params = module_desc->params;
+    auto it_dir = params.find("model_path");
+    if (it_dir == params.end()) {
+        std::cerr << "TextEmbeddingModule[" << module_desc->name << "]: 'model_path' not found in params" << std::endl;
```

Comment:
Update based on https://github.com/xipingyan/openvino.genai/pull/15, suggest to use GENAI_ERR